### PR TITLE
Added inspector for bounds control

### PR DESCRIPTION
**Pull request by thalbern**
## Overview
Adds custom inspector for bounds control that will render the scriptable object configurations as foldouts.
Following @Zee2 s advise scriptables that aren't linked will now be created inside the control and can be configured as an inlined local version of the visual configuration. Configurations that are local can be edited in place.
Configurations that are linked to a scriptable asset (shared scriptables) are rendered greyed out in the foldout and can't be edited directly in the control to make sure users won't accidentally modify shared configurations.

The foldout code has been moved into InspectorUIUtility so it can be shared across other UX elements using scriptables. 

## Changes
- Fixes: #7316 

image

### CODE REVIEW

File: Assets/MRTK/SDK/Experimental/Features/UX/BoundsControl/BoundsControlInspector.cs [hunk 1]
```diff
@@ -0,0 +1,158 @@
+﻿using Microsoft.MixedReality.Toolkit.Input;
```
Comment (keveleigh) [inline Assets/MRTK/SDK/Experimental/Features/UX/BoundsControl/BoundsControlInspector.cs:1]: Missing license header

File: Assets/MRTK/Core/Inspectors/Utilities/InspectorUIUtility.cs [hunk 2]
```diff
@@ -604,5 +604,58 @@ public static int GetOptionsIndex(SerializedProperty options, string selection)
 
             return 0;
         }
+
+        /// <summary>
+        /// Draws the contents of a scriptable inline inside a foldout. Depending on if there's an actual scriptable
+        /// linked, the values will be greyed out or editable in case the scriptable is created inside the serialized object.
+        /// </summary>
+        static public bool DrawScriptableFoldout<T>(SerializedProperty scriptable, string description, bool isExpanded) where T : ScriptableObject
+        {
```
Comment (CDiaz-MS) [inline Assets/MRTK/Core/Inspectors/Utilities/InspectorUIUtility.cs:613]: This is extremely useful, tested out `DrawScriptableFoldout` for the interactable changes and it works great.

File: Assets/MRTK/SDK/Experimental/Features/UX/BoundsControl/BoundsControlInspector.cs [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+            {
+                EditorGUI.BeginChangeCheck();
+
+                EditorGUILayout.PropertyField(targetObject);
+                EditorGUILayout.PropertyField(activationType);
+                EditorGUILayout.PropertyField(boundsOverride);
+                EditorGUILayout.PropertyField(boundsCalculationMethod);
+                EditorGUILayout.PropertyField(controlPadding);
+                EditorGUILayout.PropertyField(flattenAxis);
+
```
[the remark this answers is not in the document]
Comment (thalbern) [reply Assets/MRTK/SDK/Experimental/Features/UX/BoundsControl/BoundsControlInspector.cs]: i left one of them intentional to separate the smoothing section. now thinking about it i should probably remove that header definition in the bounds control file and add it in the inspector instead

**Pull request merged**